backend/tests/integration/tests/permissions/test_cc_pair_permissions.py

```diff
@@ -0,0 +1,165 @@
+from tests.integration.common_utils.managers.cc_pair import CCPairManager
+from tests.integration.common_utils.managers.connector import ConnectorManager
+from tests.integration.common_utils.managers.credential import CredentialManager
+from tests.integration.common_utils.managers.user import TestUser
+from tests.integration.common_utils.managers.user import UserManager
+from tests.integration.common_utils.managers.user_group import UserGroupManager
+
+
+def test_cc_pair_permissions(reset: None) -> None:
+    # Creating an admin user (first user created is automatically an admin)
+    admin_user: TestUser = UserManager.create(name="admin_user")
+
+    # Creating a curator
+    curator: TestUser = UserManager.create(name="curator")
+
+    # Creating a user group
+    user_group_1 = UserGroupManager.create(
+        name="curated_user_group",
+        user_ids=[curator.id],
+        cc_pair_ids=[],
+        user_performing_action=admin_user,
+    )
+    UserGroupManager.wait_for_sync(
+        user_groups_to_check=[user_group_1], user_performing_action=admin_user
+    )
+    # setting the user as a curator for the user group
+    UserGroupManager.set_user_to_curator(
+        test_user_group=user_group_1,
+        user_to_set_as_curator=curator,
+        user_performing_action=admin_user,
+    )
+
+    # Creating another user group that the user is not a curator of
+    user_group_2 = UserGroupManager.create(
+        name="uncurated_user_group",
+        user_ids=[curator.id],
+        cc_pair_ids=[],
+        user_performing_action=admin_user,
+    )
+    UserGroupManager.wait_for_sync(
+        user_groups_to_check=[user_group_1], user_performing_action=admin_user
+    )
+
+    # Create a credentials that the curator is and is not curator of
+    connector_1 = ConnectorManager.create(
+        name="curator_owned_connector",
+        source=DocumentSource.CONFLUENCE,
+        groups=[user_group_1.id],
+        is_public=False,
+        user_performing_action=admin_user,
+    )
+    # currently we dont enforce permissions at the connector level
+    # pending cc_pair -> connector rework
+    # connector_2 = ConnectorManager.create(
+    #     name="curator_visible_connector",
+    #     source=DocumentSource.CONFLUENCE,
+    #     groups=[user_group_2.id],
+    #     is_public=False,
+    #     user_performing_action=admin_user,
+    # )
+    credential_1 = CredentialManager.create(
+        name="curator_owned_credential",
+        source=DocumentSource.CONFLUENCE,
+        groups=[user_group_1.id],
+        curator_public=False,
+        user_performing_action=admin_user,
+    )
+    credential_2 = CredentialManager.create(
+        name="curator_visible_credential",
+        source=DocumentSource.CONFLUENCE,
+        groups=[user_group_2.id],
+        curator_public=False,
+        user_performing_action=admin_user,
+    )
+
+    # END OF HAPPY PATH
+
+    """
+    Curators should not be able to:
```

Comment:
nit: prefer to leave the comment as to what each of the below cases is testing right next to the actual code e.g.:

```
    """Tests for things Curators should not be able to do"""

    # Curator shouldn't be able to create a public cc pair
    with pytest.raises(HTTPError):
        CCPairManager.create(
            connector_id=connector_1.id,
            credential_id=credential_1.id,
            name="invalid_cc_pair_1",
            groups=[user_group_1.id],
            is_public=True,
            user_performing_action=curator,
        )
```

This makes it so the reader doesn't have to look / scroll up to figure out what's supposed to be going on

Reply:
Similar feelings in other places, although this is a bit more of a personal preference thing